lib/replicate/checkpointer.js

```diff
@@ -64,13 +94,43 @@ Checkpointer.prototype.updateSource = function (checkpoint) {
     });
 };
 
+var comparisons = {
+  "undefined": function(targetDoc, sourceDoc) {
+    // This is the previous comparison function
+    if (collate(targetDoc.last_seq, sourceDoc.last_seq) === 0) {
+      return sourceDoc.last_seq;
+    }
+
+    return 0;
+  },
+  "2": function(targetDoc, sourceDoc) {
+    // This is the comparison function ported from CouchDB
+    return compareReplicationLogs(sourceDoc, targetDoc).last_seq;
+  }
+};
+
 Checkpointer.prototype.getCheckpoint = function () {
   var self = this;
   return self.target.get(self.id).then(function (targetDoc) {
     return self.src.get(self.id).then(function (sourceDoc) {
-      if (collate(targetDoc.last_seq, sourceDoc.last_seq) === 0) {
-        return sourceDoc.last_seq;
+      // TODO: Discuss the 'return 0's here
+      // can we do something better?
+      if(targetDoc.replication_id_version !==
+         sourceDoc.replication_id_version) {
+        return 0;
+      }
+
+      var version;
+      if(targetDoc.replication_id_version) {
+        version = targetDoc.replication_id_version.toString();
+      } else {
+        version = "undefined";
       }
+
+      if(comparisons[version]) {
```

Comment:
Just a personal style: I would prefer `if(version in comparisons)` because I think it reads a little better.